programs/compliant_token_template.leo

```diff
@@ -75,23 +79,202 @@ program compliant_token_template.aleo {
     // Trees deeper than this value are not supported.
     const MAX_TREE_DEPTH: u32 = 15u32;
 
+    const MULTISIG_OP_UPGRADE: u16 = 1u16;
+    const MULTISIG_OP_UPDATE_WALLET_ID: u16 = 2u16;
+    const MULTISIG_OP_UPDATE_ROLE: u16 = 4u16;
+    const MULTISIG_OP_MINT_PUBLIC: u16 = 8u16;
+    const MULTISIG_OP_BURN_PUBLIC: u16 = 16u16;
+    const MULTISIG_OP_MINT_PRIVATE: u16 = 32u16;
+    const MULTISIG_OP_BURN_PRIVATE: u16 = 64u16;
+    const MULTISIG_OP_SET_PAUSE_STATUS: u16 = 128u16;
+    mapping wallet_id_to_role: address => u16;
+
+
+    struct CompliantTokenMultisigOp {
+        op: u16,
+
+        // Only used for MULTISIG_OP_UPDATE_ROLE, MULTISIG_OP_MINT_PUBLIC, MULTISIG_OP_BURN_PUBLIC and MULTISIG_OP_UPDATE_WALLET_ID
+        user: address,
+
+        // Only used for MULTISIG_OP_MINT_PUBLIC and MULTISIG_OP_BURN_PUBLIC
+        amount: u128,
+
+        // Only used for MULTISIG_OP_UPDATE_ROLE and MULTISIG_OP_UPDATE_WALLET_ID
+        role: u16,
+
+        // Only used for MULTISIG_OP_SET_PAUSE_STATUS
+        pause_status: bool,
+
+        // Only used for MULTISIG_OP_UPGRADE
+        edition: u16,
+        checksum: [u8; 32],
+    }
+
+    struct PrivateCompliantTokenMultisigOp {
+        op: u16,
+        user: address,
+        amount: u128,
+    }
+
+    struct CompliantTokenMultisigSaltedOp {
```

Comment:
any reason you didn't include `salt` in `CompliantTokenMultisigOp`? I would call it nonce